From d368f5fdfe88e4af0e87a0fe3e022bc2b7124af8 Mon Sep 17 00:00:00 2001
From: Parasuraman Ramalingam <x0151836@ti.com>
Date: Tue, 1 Mar 2011 14:14:33 +0530
Subject: OMAP4: DSS: Fix for interlace video rotation

In the display controller plane setup, the height and width for the
interlaced videos is set without considering the rotation values.
Because of this, 1920*1080i videos were not playing properly when the
rotation is set to 90 or 270 degrees.

This patch sets the height and width values based on rotation.
Tested on Blaze ES2.1 with different resolution videos and different
values of rotation.

Change-Id:I7d28e5c0e63fd26b274139481e5b7ef40baf8b04
Signed-off-by: Parasuraman Ramalingam <x0151836@ti.com>
Signed-off-by: Srinivas Pulukuru <srinivas.pulukuru@ti.com>
---
 drivers/media/video/omap/omap_vout.c |   28 ++++++----------------------
 drivers/video/omap2/dss/dispc.c      |   16 +++++++++++++++-
 2 files changed, 21 insertions(+), 23 deletions(-)

Index: kernel/drivers/media/video/omap/omap_vout.c
===================================================================
--- kernel.orig/drivers/media/video/omap/omap_vout.c	2011-04-05 23:40:04.000000000 +0900
+++ kernel/drivers/media/video/omap/omap_vout.c	2011-04-05 23:53:56.000000000 +0900
@@ -1307,7 +1307,7 @@
 		    !(ovl->info.field & OMAP_FLAG_IDEV) &&
 		    (irq == DISPC_IRQ_FRAMEDONE ||
 		     irq == DISPC_IRQ_FRAMEDONE2)) {
-			 /* in this case, the upper halg of the frame would be
+			 /* In this case, the upper half of the frame would be
 			  * rendered and the lower one would be ignored.
 			  */
 		}
